internal/querycoord/index_checker.go

```diff
@@ -192,13 +193,20 @@ func (ic *IndexChecker) checkIndexLoop() {
 				validHandoffReq, collectionInfo := ic.verifyHandoffReqValid(segmentInfo)
 				if validHandoffReq && Params.QueryCoordCfg.AutoHandoff {
 					indexInfo, err := getIndexInfo(ic.ctx, segmentInfo, collectionInfo.Schema, ic.rootCoord, ic.indexCoord)
-					if err != nil {
+					if err == nil {
+						// if index exist or not enableIndex, ready to load
+						segmentInfo.IndexInfos = indexInfo
+						ic.enqueueIndexedSegment(segmentInfo)
+						break
+					}
```

Comment:
If feels weird to put normal branch in if block while left the error route below

Reply:
> If feels weird to put normal branch in if block while left the error route below

The error just means that the index has not been built, not that there is an error when running.